### ISSUE TITLE
Support nested packages

### ISSUE THREAD

**Issue by likern** [mmazzarolo/react-native-monorepo-tools#5]
If I try to move packages **one level deep**, ie instead of `packages/app` and `packages/mobile` using `packages/examples/app` and `packages/examples/mobile` I see error when try to run application, saying can't find `app` package.

I think it doesn't look recursively under `packages/` directory.
This might be useful if I want to distribute package (not application) and test it under multiple applications.

```
- packages/
-- examples/
--- app/
--- mobile-0.64.0/
--- mobile-0.65.1/
--- mobile-0.66.rc3/
-- calendar/ <--- Actual package I want to publish, not app
```

**Comment by mmazzarolo** [comment]
Hmm... 🤔 I might be wrong here, but I'm not seeing how it would work with [Yarn workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces/) unless you specify each workspace in the root package.json.

**Comment by likern** [comment]
Yes, if it's required I can directly specify packages. I haven't tested, but I think _recursive search_ should work
```json
"packages": [
  "packages/**"
],
```
I suggest now the problem in `monorepo-tools`, but again not sure.

**Comment by likern** [comment]
Also I've experienced this error while trying adding web support
![Снимок экрана от 2021-09-30 20-55-36](https://user-images.githubusercontent.com/842306/135507112-4e641fab-8f51-463f-aa8b-381cbb7d75f9.png)

I've used one of the latest `yarn classic` versions. This solution from [stackoverflow](https://stackoverflow.com/questions/62254089/expected-workspace-package-to-exist-for-sane) did help
```bash
yarn policies set-version 1.19.0
```

**Comment by mmazzarolo** [comment]
> Yes, if it's required I can directly specify packages. I haven't tested, but I think _recursive search_ should work
> 
> ```json
> "packages": [
>   "packages/**"
> ],
> ```
> 
> I suggest now the problem in `monorepo-tools`, but again not sure.

Not specifically the same, but if you check the Yarn issues you'll find multiple reports about problems with non-flat structures: https://github.com/yarnpkg/yarn/pull/6151. 
I suggest you trying setting each workspace in the root package and give it a try.  

(I don't have much free time now, so I can't promise I'll give be able to test this issue anytime soon :/ )
